src/Surfnet/StepupGateway/SamlStepupProviderBundle/Provider/ConnectedServiceProviders.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupGateway\SamlStepupProviderBundle\Provider;
+
+use Surfnet\StepupGateway\GatewayBundle\Service\SamlEntityService;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Exception\InvalidArgumentException;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Exception\UnknownProviderException;
+
+/**
+ * @todo discuss (im)mutability
```

Comment:
If there's no need for runtime connecting of service providers, I suppose we should opt for constructor injection.